core/primitives/src/shard_layout.rs

```diff
@@ -241,9 +339,34 @@ pub enum ShardLayoutError {
 }
 
 impl ShardLayout {
-    /* Some constructors */
-    pub fn v0_single_shard() -> Self {
-        Self::v0(1, 0)
+    /// Handy constructor for a single-shard layout, mostly for test purposes
+    pub fn single_shard() -> Self {
+        Self::multi_shard(1, 0)
+    }
+
+    /// Can be used to construct a multi-shard layout, mostly for test purposes
+    pub fn multi_shard(num_shards: NumShards, version: ShardVersion) -> Self {
```

Comment:
how about `n_shard()` in the sense of creating an N-shard layout?

Reply:
I'm not a fan tbh. How about just `new` or `new_test`?